src/org/pahappa/systems/services/IncidentServiceImpl.java

```diff
@@ -41,7 +41,36 @@ public Incident saveIncident(Incident incident) throws Exception {
 
 	@Override
 	public Incident updateIncident(Incident incident) throws Exception {
-		// TODO Auto-generated method stub
+		//Check if both the comment and title sections have been filled
+		if (incident.getTitle()==null){
+			throw new ValidationFailedException("Please enter title");
+		}else if (incident.getTitle().isEmpty()){
+			throw new ValidationFailedException("Please enter title");
+		}else if (incident.getComment()==null){
+			throw new ValidationFailedException("Please enter comment");
+		}else if (incident.getComment().isEmpty()){
+			throw new ValidationFailedException("Please enter comment");
+		}
+
+		//Searching and updating the incident
+		Incident searchResult = findIncident(incident);
+		if (searchResult == null){
+			throw new ValidationFailedException("This record does not exist");
+		}else {
+			searchResult.setTitle(incident.getTitle());
+			searchResult.setComment(incident.getComment());
+			return searchResult;
+		}
+
```

Comment:
@ttcollins I don't think this function is working. Did you test it and see it update.